app/src/main/java/com/hedvig/app/ApplicationModule.kt

```diff
@@ -374,7 +375,14 @@ val viewModelModule = module {
   viewModel { (quoteCartId: QuoteCartId?) -> RedeemCodeViewModel(quoteCartId, get(), get()) }
   viewModel { UserViewModel(get(), get(), get(), get(), get(), get()) }
   viewModel { WelcomeViewModel(get()) }
-  viewModel { SettingsViewModel(get(), get(), get()) }
+  viewModel {
+    SettingsViewModel(
+      repository = get(),
+      localeBroadcastManager = get(),
+      hAnalytics = get(),
+      clearCache = { get<ApolloClient>().apolloStore.clearAll() },
```

Comment:
Is it too late in the after noon or is this actually beneficial? With composable we only want to pass lambdas to avoid coupling, cant we do the same with view models?

Reply:
Hmm so about this I think the reason why it may not be as useful is this.

In compose, we offload this kind of work from the Composable, to the ViewModel, which is meant to be holding such logic.
In this case we would be offloading this logic from the ViewModel to the DI Module definition, which if anything would make this already humongous file even more gigantic 😅
I think in general these modules should be as dumb and straightforward as possible, like you are this Object, and you need these dependencies, so go ahead and take them.

Imagine a if this were normal Dagger and had a @Module and a @Provides and inside the fun annotated with @Provides you'd have logic about what lambda is going what instead of just acting like a dependency injection system and simply injecting the necessary dependencies that the object needs. I think it would look even more weird there, here it looks more normal since it's Koin and all of it is more free-form sort of, if that makes  sense :D 


And besides all the long text above :D 
We have the com.hedvig.app.util.apollo.CacheManager which exists to expose exactly one function called `clearCache`, so here instead you could simply inject the CacheManager and let the ViewModel call `cacheManager.clearCache`.
Which makes me think of course maybe this could be named ApolloCacheManager or something, since CacheManager technically could mean more things.

Reply:
Yes that makes sense :)

Reply:
I will rename CacheManager to NetworkCacheManager, so that its clear that its used for network cache.

Reply:
I was thinking of ApolloCacheManager since technically we could be referring to the OkHttp network cache or the cache that Coil has with the names `CacheManager` or `NetworkCacheManager`.
Because both of those two are also caching responses from the network and we use them on our project. But we usually don't really touch those so maybe it's fine, but if we do in the future it might get a bit confusing :D